packages/paste-core/components/button/src/index.tsx

```diff
@@ -147,20 +177,55 @@ const getButtonComponent = (variant: ButtonVariants): React.FunctionComponent<Di
 // memo
 const Button = React.forwardRef<HTMLButtonElement, ButtonProps>((props, ref) => {
   const {size, variant, children, disabled, loading, ...rest} = props;
+  const [hovered, setHovered] = React.useState(false);
+  const arrowIconStyles = useSpring({
+    translateX: hovered ? '4px' : '0px',
+    config: {
+      mass: 0.1,
+      tension: 275,
+      friction: 16,
+    },
+  });
+
+  const smartDefaultSize = React.useMemo(() => {
+    return getButtonSize(variant, children, size);
+  }, [size, variant, children]);
 
-  handlePropValidation(props);
+  handlePropValidation({...props, size: smartDefaultSize});
 
   const buttonState = getButtonState(disabled, loading);
   const showLoading = buttonState === 'loading';
   const showDisabled = buttonState !== 'default';
   const ButtonComponent = getButtonComponent(variant);
-  const smartDefaultSize = React.useMemo(() => {
-    return getButtonSize(variant, children, size);
-  }, [size, variant, children]);
+  // Automatically inject AnchorForwardIcon for link's dressed as buttons when possible
+  const injectIconChildren =
+    props.as === 'a' && props.href != null && typeof children === 'string' && variant !== 'reset' ? (
+      <>
+        {children}
+        <AnimatedBox style={arrowIconStyles}>
+          <ArrowForwardIcon decorative />
+        </AnimatedBox>
+      </>
+    ) : (
+      children
+    );
 
   return (
     <ButtonComponent
+      {...(rest.href != null ? secureExternalLink(rest.href) : null)}
       {...rest}
+      onMouseEnter={(event) => {
+        if (typeof rest.onMouseEnter === 'function') {
+          rest.onMouseEnter(event);
+        }
+        setHovered(true);
+      }}
+      onMouseLeave={(event) => {
+        if (typeof rest.onMouseLeave === 'function') {
+          rest.onMouseLeave(event);
+        }
+        setHovered(false);
+      }}
```

Comment:
Performance question. Is this way any more performant than the `useHover` hook thats used in Disclosure? https://github.com/twilio-labs/paste/blob/main/packages/paste-core/components/disclosure/src/utils.ts#L43

That might help answer the question of if we need to pull that out into a hooks library.

Reply:
Nice, good call out

Reply:
I haven't measured the performance, but my hunch is that they're reasonably equivalent. 
Some differences:
- My implementation used mouseEnter rather than mouseOver, which has a different behavior (https://www.w3schools.com/jquery/tryit.asp?filename=tryjquery_event_mouseenter_mouseover#:~:text=mouseenter%20and%20mouseover.-,The%20mouseover%20event%20triggers%20when%20the%20mouse%20pointer%20enters%20the,moved%20over%20the%20div%20element.)
- the `useHover` hook works with `ref` which may be cleaner for where you need the information about mouseOver on nested components defined in other files or files you don't control. For example, if the `mouseOver` event is not passed to a component that you are consuming, this hook uses ref to add the event via JS rather than react's event system (https://reactjs.org/docs/events.html)

I think both approaches are fine for now.